File: backend/src/main/java/com/back/back9/domain/wallet/repository/WalletRepository.java
package com.back.back9.domain.wallet.repository;

import com.back.back9.domain.user.entity.User;
import com.back.back9.domain.wallet.entity.Wallet;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface WalletRepository extends JpaRepository<Wallet, Long> {
    boolean existsByUserId(Long userId);
    Optional<Wallet> findByUser(User user);
    Optional<Wallet> findByUserId(Long userId);  // Long 타입으로 변경
}

File: backend/src/main/kotlin/com/back/back9/domain/wallet/repository/WalletRepository.kt
package com.back.back9.domain.wallet.repository

import com.back.back9.domain.user.entity.User
import com.back.back9.domain.wallet.entity.Wallet
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WalletRepository : JpaRepository<Wallet, Long> {
    fun existsByUserId(userId: Long): Boolean
    fun findByUser(user: User): Wallet?
    fun findByUserId(userId: Long): Wallet?
}
